rsksmart/lumino
# Feature/rlp 849/receive lc messages design

This PR intends to be the first Implementation of the Rif Comms `LightClientNode`

It also fixes some tests to use the current implementation of rif comms pubsub

Manual testing Steps: 
- A Full node transport and a Light client transport are initialized (addresses 0x1 and 0x2) 
- full node transport subscribes to 0x1
- light transport subscribes to 0x2
- the subscription shows the same peer id for each one (See *1)
- the sender.py script sends a message to 0x1
- the message is received sometimes by full node transport and some times by light transport 
- the sender.py script sends a message to 0x2 and the same occurs

This table represents the behaviour vaguely

|  sender calls with | first time  | second time   |  restart all  | restart lumino (subscribe again)   |
|---|---|---|---|---|
| lc  | lc   | lc  | lc   | fn  |
| fn  | lc   | lc  | lc   | fn  |

Notes that represent incorrect behaviour: 

- each message is received twice
- each message is received on the wrong topic, for example, if sender sends a message with fn address, it doesnt make sense to receive it on the lc topic

(*1)

```
createTopicWithRskAddress {"address":"0x138aF366e0ED7Cc4b9747a935D1b5F75A86b9D83","exclusive":false} 
locatePeerID "0x138aF366e0ED7Cc4b9747a935D1b5F75A86b9D83" 
connectToCommunicationsNode {"address":"0x138aF366e0ED7Cc4b9747a935D1b5F75A86b9D83","exclusive":false}
createTopicWithRskAddress {"address":"0x2Cd53d83E5ca1F3c66EDC7bf307139c6B1989C9a","exclusive":false} 
locatePeerID "0x2Cd53d83E5ca1F3c66EDC7bf307139c6B1989C9a" 
connectToCommunicationsNode {"address":"0x2Cd53d83E5ca1F3c66EDC7bf307139c6B1989C9a","exclusive":false}
Adding RSKADDRESS PEER= 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA  : RSKADDRESS= 0x138aF366e0ED7Cc4b9747a935D1b5F75A86b9D83
Adding RSKADDRESS PEER= 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA  : RSKADDRESS= 0x2Cd53d83E5ca1F3c66EDC7bf307139c6B1989C9a
address 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA
 - New subscription to 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA
address 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA
 - New subscription to 16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA

```
